core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,279 @@
-	//   - manage trigger subscriptions
-	//   - process incoming trigger events and related metadata
-	return nil, nil
+const defaultSendChannelBufferSize = 1000
+
+const TriggerType = "web-trigger@1.0.0"
+
+var webapiTriggerInfo = capabilities.MustNewCapabilityInfo(
+	TriggerType,
+	capabilities.CapabilityTypeTrigger,
+	"A trigger to start workflow execution from a web api call",
+)
+
+type Input struct {
+}
+type Config struct {
+	AllowedSenders []string                 `toml:"allowedSenders"`
+	AllowedTopics  []string                 `toml:"allowedTopics"`
+	RateLimiter    common.RateLimiterConfig `toml:"rateLimiter"`
+	// RequiredParams is advisory to the web trigger message sender it is not enforced.
+	RequiredParams []string `toml:"requiredParams"`
+}
+
+type webapiTrigger struct {
+	allowedSenders map[string]bool
+	allowedTopics  map[string]bool
+	ch             chan<- capabilities.TriggerResponse
+	config         Config
+	rateLimiter    *common.RateLimiter
+}
+
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	capabilities.Validator[Config, Input, capabilities.TriggerResponse]
+	connector           connector.GatewayConnector
+	lggr                logger.Logger
+	mu                  sync.Mutex
+	registeredWorkflows map[string]webapiTrigger
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	if connector == nil {
+		return nil, errors.New("missing connector")
+	}
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[Config, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// processTrigger iterates over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload webapicapabilities.TriggerRequestPayload) error {
+	// Pass on the payload with the expectation that it's in an acceptable format for the executor
+	wrappedPayload, err := values.WrapMap(payload)
+	if err != nil {
+		return fmt.Errorf("error wrapping payload %s", err)
+	}
+	topics := payload.Topics
+
+	// empty topics is error for V1
+	if len(topics) == 0 {
+		return fmt.Errorf("empty Workflow Topics")
+	}
+
+	// workflows that have matched topics
+	matchedWorkflows := 0
+	// workflows that have matched topic and passed all checks
+	fullyMatchedWorkflows := 0
+	for _, trigger := range h.registeredWorkflows {
```

Comment:
I still don't like the time complexity here from iterating over _every workflow and its topics_ on a node.

I'd prefer to see a unit test showing load testing for benchmarking, but won't block the PR over it. Could we have a ticket tracking optimizing this?

Reply:
Happy to do a ticket.  I'd love to learn how we can do load testing in unit tests.